laa_court_data_api_app/routers/laa_references.py

```diff
@@ -33,6 +33,11 @@
 
 @router.patch("/v2/laa_references/{defendant_id}/", status_code=202, responses=responses)
 async def patch_maat_unlink(defendant_id: str, request: ExternalPatchRequest):
+    if (defendant_id != str(request.defendant_id)):
+        logger.info("Mismatched_DefendantId_In_Patch_Request")
+        return JSONResponse(status_code=400, content=LaaReferencesErrorResponse(
+            error={'defendant_id': ['mismatch in ids given']}).dict())
```

Comment:
To confirm, this is validation because the CDA maat unlink endpoint needs defendant_id in the param and request body, which can lead to a mismatch. However, this is a fix around that.

Although, my belief is the duplication of defendant_id in the request is not ideal and CDA may need to change that, right?

Reply:
Correct - because it is defined in multiple locations we are ensuring that there is match and that a malicious user isnt trying to overwrite one with another